src/main/src/test/java/org/geoserver/catalog/RetypeFeatureTypeCallbackTest.java

```diff
@@ -0,0 +1,98 @@
+/* (c) 2020 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.catalog;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.xml.namespace.QName;
+import org.geoserver.data.test.MockData;
+import org.geoserver.data.test.SystemTestData;
+import org.geoserver.data.test.SystemTestData.LayerProperty;
+import org.geoserver.test.GeoServerSystemTestSupport;
+import org.geotools.data.FeatureSource;
+import org.geotools.feature.FeatureIterator;
+import org.geotools.referencing.CRS;
+import org.junit.Test;
+import org.locationtech.jts.geom.Point;
+import org.opengis.feature.Feature;
+import org.opengis.feature.type.FeatureType;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+import org.vfny.geoserver.global.GeoServerFeatureSource;
+
+/**
+ * This test makes use of mock RetypeFeatureTypeCallback implementation provided in
+ *
+ * @author ImranR
+ */
+public class RetypeFeatureTypeCallbackTest extends GeoServerSystemTestSupport {
+
+    public static final String LONG_LAT_NO_GEOM_ON_THE_FLY_LAYER = "longlat";
+    public static final QName LONG_LAT_NO_GEOM_ON_THE_FLY_QNAME =
+            new QName(
+                    MockData.DEFAULT_PREFIX,
+                    LONG_LAT_NO_GEOM_ON_THE_FLY_LAYER,
+                    MockData.DEFAULT_PREFIX);
+
+    @Override
+    protected void onSetUp(SystemTestData testData) throws Exception {
+        super.onSetUp(testData);
+        setupBasicLayer(testData);
+    }
+
+    private void setupBasicLayer(SystemTestData testData) throws IOException {

```

Comment:
This is the basic layer for a specific test, the method name is too generic.

Reply:
will change